OpenRA.Mods.Common/Projectiles/Bullet.cs

```diff
@@ -302,12 +303,12 @@ bool AnyValidTargetsInRadius(World world, WPos pos, WDist radius, Actor firedBy,
 				if (!info.ValidBounceBlockerStances.HasStance(victim.Owner.Stances[firedBy.Owner]))
 					continue;
 
-				var healthInfo = victim.Info.TraitInfoOrDefault<HealthInfo>();
-				if (healthInfo == null)
+				var activeShapes = victim.TraitsImplementing<HitShape>().Where(Exts.IsTraitEnabled).ToArray();
```

Comment:
This can be made more efficient by removing the `ToArray` and the `if (!activeShapes.Any())`.  The `activeShapes.Any(i => ...)` check can then return before having to fully evaluate the enumerable.